packages/bridging/src/BridgingSdk/BridgingSdk.ts

```diff
@@ -136,7 +111,19 @@ export class BridgingSdk {
    * @param params
    */
   async getBuyTokens(params: BuyTokensParams): Promise<GetProviderBuyTokens> {
-    return this.provider.getBuyTokens(params)
+    const providerId = this.provider.info.dappId
+
+    const cacheKey = `${providerId}-${params.buyChainId}-${params.sellChainId}-${params.sellTokenAddress?.toLowerCase()}`
+
+    if (this.cacheConfig.enabled && this.buyTokensCache.get(cacheKey)) {
+      return this.buyTokensCache.get(cacheKey) as GetProviderBuyTokens
+    }
+
+    const result = await this.provider.getBuyTokens(params)
+    if (this.cacheConfig.enabled) {
+      this.buyTokensCache.set(cacheKey, result, this.cacheConfig.buyTokensTtl)
```

Comment:
Maybe we can put `ttl` into `TTLCache` constructor? With that we won't need to pass it as a parameter every time when you call `get` or `set`